src/Umbraco.Web.UI.Client/src/views/propertyeditors/contentpicker/contentpicker.controller.js

```diff
@@ -215,26 +232,69 @@ function contentPickerController($scope, dialogService, entityResource, editorSt
 
     //load current data
     var modelIds = $scope.model.value ? $scope.model.value.split(',') : [];
+
     entityResource.getByIds(modelIds, entityType).then(function (data) {
 
-        //Ensure we populate the render model in the same order that the ids were stored!
         _.each(modelIds, function (id, i) {
             var entity = _.find(data, function (d) {                
                 return d.id == id;
             });
-           
+        
             if (entity) {
-                entity.icon = iconHelper.convertFromLegacyIcon(entity.icon);
-                $scope.renderModel.push({ name: entity.name, id: entity.id, icon: entity.icon, path: entity.path });
+                // get url for content and media items
+                if(entityType !== "Member") {

```

Comment:
Practically identical code as lines 197-205 apart from the object you are passing in & then updating the property on. Perhaps could be made into a function to avoid code duplication @madsrasmussen ?